libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.helpers.spec.ts

```diff
@@ -0,0 +1,265 @@
+      const amount = calculateAmount({
+        creditsDocumentUnitPrice,
+        massIdCertificateValue: new BigNumber(1500),
+        massIdPercentage: '30.1234567892',
+        participantAmount: '3.223669',
+      });
+
+      expect(amount.toString()).toEqual('2366.408854');
+    });
+
+    it('should handle undefined participantAmount', () => {
+      const amount = calculateAmount({
+        creditsDocumentUnitPrice,
+        massIdCertificateValue: new BigNumber(100),
+        massIdPercentage: '10',
+        participantAmount: undefined,
+      });
+
+      expect(amount).toBe('52.3');
+    });
+  });
+
+  describe('calculateRemainder', () => {
+    it('should return the correct remainder of the credit', () => {
+      const massIdTotalValue = new BigNumber(2500);
+      const actors: ActorsByActorType = new Map();
+
+      // eslint-disable-next-line @typescript-eslint/naming-convention, no-underscore-dangle, @typescript-eslint/no-unused-vars
+      for (const _iterator of Array.from({ length: 3 })) {
+        actors.set(
+          `${random<RewardsDistributionActorType>()}-${faker.string.uuid()}`,
+          {
+            ...random<RewardsDistributionActor>(),
+            amount: '12.503982',
+            percentage: '33.333333',
+          },
+        );
+      }
+
+      const { amount, percentage } = calculateRemainder({
+        actors,
+        creditsDocumentUnitPrice,
+        massIdTotalValue,
+      });
+
+      expect(amount.toString()).toEqual('13037.488054');
+      expect(percentage.toString()).toEqual('0.000001');
+    });
+  });
+
+  describe('addParticipantRemainder', () => {
+    it('should sum remainder value to participant amount and remainderPercentage', () => {
+      const actors: ActorsByActorType = new Map();
+      const remainder = {
+        amount: new BigNumber('3.543984'),
+        percentage: new BigNumber('0.027105'),
+      };
+
+      actors.set(MethodologyActorType.NETWORK, {
+        ...random<RewardsDistributionActor>(),
+        actorType: RewardsDistributionActorType.NETWORK,
+        amount: '8.023876',
+      });
+
+      addParticipantRemainder({
+        actors,
+        remainder,
+      });
+
+      expect(actors.get(MethodologyActorType.NETWORK)).toMatchObject({
+        amount: '11.56786',
+      });
+    });
+
+    it('should do nothing if NETWORK actor is not present', () => {
+      const actors: ActorsByActorType = new Map();
+
+      actors.set('OTHER-1', {
+        actorType: RewardsDistributionActorType.HAULER,
+        amount: '1',
+        participant: { id: '1', name: 'A' },
+        percentage: '1',
+      });
+      const remainder = {
+        amount: new BigNumber(5),
+        percentage: new BigNumber(5),
+      };
+
+      addParticipantRemainder({ actors, remainder });
+      expect(actors.get('OTHER-1')?.amount).toBe('1');
+    });
+  });
+
+  describe('aggregateCertificateRewards', () => {
+    it('should add amount, actorType and name to participant', () => {
```

Comment:
I think it would be nice to have a test that has the same participant and actor type in two different certificates, so their values need to be summed